app/src/main/java/org/oppia/android/app/options/AppLanguageActivityPresenter.kt

```diff
@@ -1,20 +1,33 @@
 package org.oppia.android.app.options
 
+import android.content.Intent
 import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
 import org.oppia.android.R
 import org.oppia.android.app.activity.ActivityScope
+import org.oppia.android.databinding.AppLanguageActivityBinding
 import javax.inject.Inject
 
 /** The presenter for [AppLanguageActivity]. */
 @ActivityScope
 class AppLanguageActivityPresenter @Inject constructor(private val activity: AppCompatActivity) {
   private lateinit var prefSummaryValue: String
 
-  fun handleOnCreate(prefKey: String, prefSummaryValue: String) {
-    activity.setContentView(R.layout.app_language_activity)
-    setLanguageSelected(prefSummaryValue)
+  fun handleOnCreate(prefKey: String, prefValue: String) {
+    val binding: AppLanguageActivityBinding = DataBindingUtil.setContentView(
+      activity,
+      R.layout.app_language_activity,
+    )
+    binding.appLanguageToolbar.setNavigationOnClickListener {
+      val intent = Intent().apply {
+        putExtra(MESSAGE_APP_LANGUAGE_ARGUMENT_KEY, prefSummaryValue)
+      }
+      (activity as AppLanguageActivity).setResult(REQUEST_CODE_APP_LANGUAGE, intent)
+      activity.finish()
+    }
+    setLanguageSelected(prefValue)
     if (getAppLanguageFragment() == null) {
-      val appLanguageFragment = AppLanguageFragment.newInstance(prefKey, prefSummaryValue)
+      val appLanguageFragment = AppLanguageFragment.newInstance(prefKey, prefValue)
```

Comment:
why are we using here prefValue instead of prefSummaryValue

Reply:
I was using #3462 (Merge audio_language_fragment into single xml file) as reference and it also had a similar change.

Reply:
ok I get it now we are calling this method in the overidden activity and then we want to use the value we passed form the saved instance state and we also have a global value in presenter so that we can use it at other places also.